From e6828c8f629fd52d7b065c45d52b6bd04acd616f Mon Sep 17 00:00:00 2001
From: Tom de Vries <tdevries@suse.de>
Date: Fri, 6 Jun 2025 10:14:47 +0200
Subject: [PATCH] [gdb/testsuite] Fix timeout in
 gdb.multi/attach-while-running.exp

With test-case gdb.multi/attach-while-running.exp usually I get:
...
(gdb) run &^M
Starting program: attach-while-running ^M
(gdb) PASS: $exp: run &
[Thread debugging using libthread_db enabled]^M
Using host libthread_db library "/lib64/libthread_db.so.1".^M
add-inferior^M
[New inferior 2]^M
Added inferior 2 on connection 1 (native)^M
(gdb) PASS: $exp: add-inferior
...
or:
...
(gdb) run &
Starting program: attach-while-running
(gdb) PASS: $exp: run &
add-inferior
[New inferior 2]
Added inferior 2 on connection 1 (native)
(gdb) PASS: $exp: add-inferior
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
...
but sometimes I run into:
...
(gdb) run &
Starting program: attach-while-running
(gdb) PASS: $exp: run &
add-inferior
[New inferior 2]
Added inferior 2 on connection 1 (native)
(gdb) [Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
FAIL: $exp: add-inferior (timeout)
...

Fix this by using -no-prompt-anchor.

Tested on x86_64-linux.
---
 gdb/testsuite/gdb.multi/attach-while-running.exp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdb/testsuite/gdb.multi/attach-while-running.exp b/gdb/testsuite/gdb.multi/attach-while-running.exp
index ca4fa635467..b67e03d87e8 100644
--- a/gdb/testsuite/gdb.multi/attach-while-running.exp
+++ b/gdb/testsuite/gdb.multi/attach-while-running.exp
@@ -49,7 +49,7 @@ proc do_test {} {
     }
 
     gdb_test -no-prompt-anchor "run &"
-    gdb_test "add-inferior" "Added inferior 2 on connection 1 .*"
+    gdb_test -no-prompt-anchor "add-inferior" "Added inferior 2 on connection 1 .*"
     gdb_test "inferior 2" "Switching to inferior 2 .*"
 
     set spawn_id [spawn_wait_for_attach $::binfile]

base-commit: 0293f33bd589d8d3a4c9cbc889934975e5fe50f2
-- 
2.43.0

